strawberryfields/backends/tfbackend/circuit.py

```diff
@@ -803,185 +726,169 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
             batch_offset = 0
             batch_size = 1
 
-        with self.graph.as_default():
-            phi = tf.cast(phi, ops.def_type)
-            phi = self._maybe_batch(phi)
+        phi = tf.cast(phi, ops.def_type)
```

Comment:
Similar thoughts as on `measure_homodyne`: might be worth splitting up this function into using auxiliary functions such that the logical parts are well separated and bugs can hide less easily.

Reply:
Agree!

Reply:
Worth mentioning that one of the major things people didn't like about TF1.0 was that it was very hard to debug. Previously, splitting up this function would have made little difference for tracking down different bugs. 

Also, some of the functionality here only appears here, and it would possibly be artificial to split into separate functions